src/Agent.Sdk/Util/TeeUtil.cs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Util
+{
+    public static class TeeUtil
+    {
+        private static readonly string TeeTempDir = "tee_temp_dir";
+
+        private static readonly string TeeUrl = "https://vstsagenttools.blob.core.windows.net/tools/tee/14_135_0/TEE-CLC-14.135.0.zip";
+
+        public static void DownloadTeeIfAbsent(
+            string agentHomeDirectory,
+            string agentTempDirectory,
+            int providedDownloadRetryCount,
+            Action<string> debug,
+            CancellationToken cancellationToken
+        ) {
+            if (Directory.Exists(GetTeePath(agentHomeDirectory)))
+            {
+                return;
+            }
+
+            int downloadRetryCount = Math.Min(Math.Max(providedDownloadRetryCount, 3), 10);
```

Comment:
Let's move constants out - at least to the class level

Reply:
Done